lang/cs/Org.Apache.REEF.IO/FileSystem/AzureDataLake/AzureDataLakeFileSystem.cs

```diff
@@ -173,14 +173,23 @@ public void CreateDirectory(Uri directoryUri)
             }
         }
 
+        /// <summary>
+        /// Checks if uri is a directory uri.
+        /// </summary>
+        /// <param name="uri">uri of the directory/file</param>
+        /// <returns>true if uri is for a directory else false</returns>
+        public bool IsDirectory(Uri uri)
+        {
+            return Exists(uri) && _adlsClient.GetDirectoryEntry(uri.AbsolutePath).Type == DirectoryEntryType.DIRECTORY;
```

Comment:
>Exists(uri) && [](start = 19, length = 14)

Call Exists doubles the number of network calls. Can we just rely on the GetDirectoryEntry alone?

Reply:
removed

---
In reply to: [194554633](https://github.com/apache/reef/pull/1473#discussion_r194554633) [](ancestors = 194554633)

Reply:
This fixes the double network call but the catch all could hide simple exceptions like a network error. 

---
In reply to: [194580374](https://github.com/apache/reef/pull/1473#discussion_r194580374) [](ancestors = 194580374,194554633)

Reply:
change reverted

---
In reply to: [195218945](https://github.com/apache/reef/pull/1473#discussion_r195218945) [](ancestors = 195218945,194580374,194554633)

Reply:
Why was the change reverted?

Reply:
_adlsClient.GetDirectoryEntry(uri.AbsolutePath) throws exception if uri not present

---
In reply to: [195886246](https://github.com/apache/reef/pull/1473#discussion_r195886246) [](ancestors = 195886246)

Reply:
Can't that exception be caught then and return false? So if you look at the source for adls client, [https://github.com/Azure/azure-data-lake-store-net/blob/master/AdlsDotNetSDK/ADLSClient.cs](https://github.com/Azure/azure-data-lake-store-net/blob/master/AdlsDotNetSDK/ADLSClient.cs)
CheckExists internally call GetDirectoryEntry, so you are calling it twice.


---
In reply to: [196918926](https://github.com/apache/reef/pull/1473#discussion_r196918926) [](ancestors = 196918926,195886246)

Reply:
fixed

---
In reply to: [197270432](https://github.com/apache/reef/pull/1473#discussion_r197270432) [](ancestors = 197270432,196918926,195886246)